esmvalcore/_citation.py

```diff
@@ -0,0 +1,218 @@
+            info_urls.append(_make_info_url(url_prefix))
+            json_urls.append(_make_json_url(url_prefix))
+        if reference_attr:
+            # get recipe citation tags
+            if item.identifier.namespace.prefix == 'recipe':
+                product_tags.extend(reference_attr)
+            # get diagnostics citation tags
+            elif item.get_attribute('attribute:script_file'):
+                product_tags.extend(reference_attr)
+            else:
+                info_urls += list(reference_attr)
+
+    _save_citation_info(product_name, product_tags, json_urls, info_urls)
+
+
+def _save_citation_info(product_name, product_tags, json_urls, info_urls):
+    citation_entries = [ESMVALTOOL_PAPER]
+
+    # save CMIP6 url_info, if any
+    # save any refrences info that is not related to recipe or diagnostics
+    title = [
+        "Some citation information are found, "
+        "which are not mentioned in the recipe or diagnostic."
+    ]
+    if info_urls:
+        with open(f'{product_name}_data_citation_info.txt', 'w') as file:
+            file.write('\n'.join(title + list(set(info_urls))))
+
+    # convert json_urls to bibtex entries
+    for json_url in json_urls:
+        cmip_citation = _collect_cmip_citation(json_url)
+        if cmip_citation:
+            citation_entries.append(cmip_citation)
+
+    # convert tags to bibtex entries
+    if REFERENCES_PATH and product_tags:
+        tags = _extract_tags(product_tags)
+        for tag in tags:
+            citation_entries.append(_collect_bibtex_citation(tag))
+
+    with open(f'{product_name}_citation.bibtex', 'w') as file:
+        file.write('\n'.join(citation_entries))
+
+
+def _extract_tags(tags):
+    """
+    Extract tags that are recorded by provenance as
+    for example, "['acknow_project', 'acknow_author']".
+    """
+    pattern = re.compile(r'\w+')
+    return list(set(pattern.findall(str(tags))))
+
+
+def _get_response(url):
+    """Return information from CMIP6 Data Citation service in json format."""
+    json_data = False
+    if url.lower().startswith('https'):
+        try:
+            response = requests.get(url)
+            if response.status_code == 200:
+                json_data = response.json()
+            else:
+                logger.warning('Error in the CMIP6 citation link: %s', url)
+        except IOError:
+            logger.info(
+                'No network connection,'
+                'unable to retrieve CMIP6 citation information'
+            )
+    return json_data
+
+
+def _json_to_bibtex(data):
+    """Make a bibtex entry from CMIP6 Data Citation json data."""
+    url = 'url not found'
+    title = data.get('titles', ['title not found'])[0]
+    publisher = data.get('publisher', 'publisher not found')
+    year = data.get('publicationYear', 'publicationYear not found')
+    authors = 'creators not found'
+    doi = 'doi not found'
+
+    if data.get('creators', ''):
```

Comment:
```suggestion
    if data.get('creators'):
```
because the default return value of `dict.get` is `None`, which also evaluates to `False` in the `if` statement. Maybe writing `if 'creators' in data` would be even more clear, since the case where `creators` is an empty list is already covered on line 147.